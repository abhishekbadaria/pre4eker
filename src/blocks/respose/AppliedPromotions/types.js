const types = {
  earnedBonusPoints: "Начисление баллов",
  spentBonusPoints: "Скидка балламми",
  deliveryDiscount: "Скидка на доставку",
  message: "Сообщение",
  issuedCoupon: "Выданный промокод",
  discount: "Скидка",
};

export default types;