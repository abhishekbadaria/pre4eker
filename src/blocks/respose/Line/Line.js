import React from "react";
import PropTypes from "prop-types";
import stylesShared from "src/blocks/respose/responseShared.css";

import AppliedPromotion from "../AppliedPromotion/AppliedPromotion";

const Line = ({ lineInfo }) => {
  const {
    product,
    quantity,
    basePricePerItem,
    discountedPriceOfLine,
    status,
    lineNumber,
    lineId,
    appliedPromotions,
    placeholders,
  } = lineInfo;

  return (
    <div>
      <div className={stylesShared.customer}>
        <div className={stylesShared.inline}>
          <div className={`${stylesShared.third}`}>
            <div className={`${stylesShared.status} ${stylesShared.neutral}`}>
              <div>№ линии: {lineNumber}</div>
              <div>ID линии: {lineId}</div>
            </div>
          </div>
          <div className={stylesShared.third}>
            <div className={stylesShared.name}>Ид продукта</div>
            <div> {product.ids[Object.keys(product.ids)[0]]}</div>
          </div>
          <div className={stylesShared.third}>
            <div className={stylesShared.name}>Количество</div>
            <div> {quantity}</div>
          </div>
        </div>
        <div className={stylesShared.inline}>
          <div className={stylesShared.half}>
            <div className={stylesShared.value}>До скидки</div>
            <div className={stylesShared.promoValue}>{basePricePerItem}</div>
          </div>
          <div className={stylesShared.half}>
            <div className={stylesShared.value}>Итого</div>
            <div className={stylesShared.promoValue}>
              {discountedPriceOfLine}
            </div>
          </div>
        </div>
      </div>
      {appliedPromotions &&
        appliedPromotions.map((promo, i) => (
          <AppliedPromotion key={`line_prmo_${i}`} promoInfo={promo} />
        ))}
    </div>
  );
};

Line.propTypes = {};

export default Line;
