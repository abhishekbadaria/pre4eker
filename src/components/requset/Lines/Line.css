.inline {
  display: flex;
  /* margin: 0 -.3em 0 -.3em; */
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;
  margin-bottom: .5rem;
}

.half {
  width: 49%;
  /* padding: 1rem .3rem; */
}

.third {
  width: 32%;
  /* padding: 0 .3rem; */
}

.quarter {
  width: 24%;
}

.h2 {
  margin: 0;
}

.h4 {
  margin: 0;
}

.lineSelector {
  padding: 0 .3rem;
  display: flex;
  flex-wrap: wrap;
  max-width: 240px;
}

.lineSelector Button {
  margin-right: -1px;
  margin-top: -1px;
}

.lineAddRemove {
  display: flex;
}

.lineAddRemove Button {
  margin-right: -1px;
}