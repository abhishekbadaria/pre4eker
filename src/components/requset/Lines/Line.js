import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";

import Input from "src/components/Input/Input";
import arrayFunctions from "../arrayFunctions";

import CustomFields from "../CustomFields/CustomFields";
import ExternalPromo from "../ExternalPromo/ExternalPromo";

import style from "./Line.css";

const Line = ({ line, lines, setLines }) => {
  const { lineId, status, productId, externalSystem, quantity, price } = line;

  const [lineState, handleChangeLineState] = useState(line);

  useEffect(() => {
    setLines(arrayFunctions.updateItem(lines, lineState));
  }, [lineState]);

  return (
    <>
      {/* <p>Линия №{line.number}</p> */}
      <div className={style.inline}>
        <Input
          label="Ид линии"
          className={style.half}
          name={`Line-id-${line.number}`}
          value={lineId}
          onChange={(e) => {
            const upDatedValue = {
              ...line,
              lineId: e.target.value,
            };
            setLines(arrayFunctions.updateItem(lines, upDatedValue));
          }}
        />

        <Input
          label="Статус линии"
          className={style.half}
          name={`Line-status-${line.number}`}
          value={status || ""}
          onChange={(e) => {
            const upDatedValue = {
              ...line,
              status: e.target.value,
            };
            setLines(arrayFunctions.updateItem(lines, upDatedValue));
          }}
        />
      </div>
      <div className={style.inline}>
        <Input
          label="Внешняя система"
          className={style.quarter}
          name={`Line-externalSystem-${line.number}`}
          value={externalSystem || "Website"}
          onChange={(e) => {
            const upDatedValue = {
              ...line,
              externalSystem: e.target.value,
            };
            setLines(arrayFunctions.updateItem(lines, upDatedValue));
          }}
        />
        <Input
          label="Идентификатор"
          className={style.quarter}
          name={`Line-id-${line.number}`}
          value={productId}
          onChange={(e) => {
            const upDatedValue = {
              ...line,
              productId: e.target.value,
            };
            setLines(arrayFunctions.updateItem(lines, upDatedValue));
          }}
        />
        <Input
          label="Количество"
          className={style.quarter}
          name={`Line-quantity-${line.number}`}
          value={quantity}
          onChange={(e) => {
            const upDatedValue = {
              ...line,
              quantity: e.target.value,
            };
            setLines(arrayFunctions.updateItem(lines, upDatedValue));
          }}
        />
        <Input
          label="Цена за 1 шт"
          className={style.quarter}
          name={`Line-price-${line.number}`}
          value={price}
          onChange={(e) => {
            const upDatedValue = {
              ...line,
              price: e.target.value,
            };
            setLines(arrayFunctions.updateItem(lines, upDatedValue));
          }}
        />
      </div>
      <CustomFields
        body={lineState}
        setBody={handleChangeLineState}
        typeOfParrent="line"
      />
      <ExternalPromo
        body={lineState}
        setBody={handleChangeLineState}
        typeOfParrent="line"
      />
    </>
  );
};

Line.propTypes = {
  line: PropTypes.object,
  lines: PropTypes.array,
  setLines: PropTypes.func,
};
Line.defaultProps = {
  line: {},
  lines: [],
  setLines: () => ({}),
};

export default Line;
