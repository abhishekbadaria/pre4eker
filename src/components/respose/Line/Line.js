import React from "react";
import PropTypes from "prop-types";
import stylesShared from "src/components/respose/responseShared.css";
import shortid from "shortid";

import AppliedPromotion from "../AppliedPromotion/AppliedPromotion";
import Placeholders from "../Placeholders/Placeholders";

const Line = ({ lineInfo }) => {
  const {
    product,
    quantity,
    basePricePerItem,
    discountedPriceOfLine,
    lineNumber,
    lineId,
    appliedPromotions,
    placeholders,
  } = lineInfo;

  let promoWithIDs;
  if (typeof appliedPromotions !== "undefined") {
    promoWithIDs = appliedPromotions.map((promo) => ({
      ...promo,
      id: shortid.generate(),
    }));
  }
  return (
    <div>
      <div className={stylesShared.customer}>
        <div className={`${stylesShared.inline} ${stylesShared.line}`}>
          <div className={`${stylesShared.third}`}>
            <div className={`${stylesShared.status} ${stylesShared.neutral}`}>
              <div>№ линии: {lineNumber}</div>
              {lineId && <div>{`ID линии: ${lineId}`}</div>}
            </div>
          </div>
          <div className={stylesShared.third}>
            <div className={stylesShared.name}>Ид продукта</div>
            {product.ids[Object.keys(product.ids)[0]]}
          </div>
          <div className={stylesShared.third}>
            <div className={stylesShared.name}>Количество</div>
            {quantity}
          </div>
        </div>
        <div className={stylesShared.inline}>
          <div className={stylesShared.half}>
            <div className={stylesShared.name}>Базовая цена</div>
            <div className={stylesShared.promoValue}>{basePricePerItem}</div>
          </div>
          <div className={stylesShared.half}>
            <div className={stylesShared.name}>Сумма по линии</div>
            <div className={stylesShared.promoValue}>
              {discountedPriceOfLine}
            </div>
          </div>
        </div>
      </div>
      {appliedPromotions &&
        promoWithIDs.map((promo) => (
          <AppliedPromotion key={promo.id} promoInfo={promo} />
        ))}
      {placeholders && <Placeholders placeholders={placeholders} />}
    </div>
  );
};

Line.propTypes = {
  lineInfo: PropTypes.object,
};
Line.defaultProps = {
  lineInfo: {},
};

export default Line;
