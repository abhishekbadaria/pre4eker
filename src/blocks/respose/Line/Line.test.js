import React from "react";
import Line from "./Line";
import { render, fireEvent, waitFor, screen, cleanup, act } from '@testing-library/react' 
import response from 'data/response'
import { exportAllDeclaration } from "@babel/types";

afterEach(cleanup)
 
const {lines} = response.order;

test(`Line has to match snapshot`, () => {
  const {asFragment, getByRole} = render(<Line lineInfo={lines[0]} />)
  expect(asFragment(<Line />)).toMatchSnapshot(); 
})