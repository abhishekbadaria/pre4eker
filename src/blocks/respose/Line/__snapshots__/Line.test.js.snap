// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line has to match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="customer"
    >
      <div
        class="inline"
      >
        <div
          class="third"
        >
          <div
            class="status neutral"
          >
            <div>
              № линии: 1
            </div>
            <div>
              ID линии: 8
            </div>
          </div>
        </div>
        <div
          class="third"
        >
          <div
            class="name"
          >
            Ид продукта
          </div>
          <div>
             474177964
          </div>
        </div>
        <div
          class="third"
        >
          <div
            class="name"
          >
            Количество
          </div>
          <div>
             13179398.15
          </div>
        </div>
      </div>
      <div
        class="inline"
      >
        <div
          class="half"
        >
          <div
            class="value"
          >
            До скидки
          </div>
          <div
            class="promoValue"
          >
            17639978.17
          </div>
        </div>
        <div
          class="half"
        >
          <div
            class="value"
          >
            Итого
          </div>
          <div
            class="promoValue"
          >
            804584.3
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="promo"
      >
        <div
          class="promoType"
        >
          Начисление баллов
        </div>
        <div
          class="inline"
        >
          <div
            class="half"
          >
            <div
              class="value"
            >
               Новая акция
            </div>
            <div
              class="line"
            >
              <div
                class="name"
              >
                Примененный промокод
              </div>
              <div
                class="value"
              >
                ABC12345
              </div>
              <div
                class="status neutral"
              >
                Из пула: Новый пул промокодов
              </div>
            </div>
            <div
              class="line"
            >
              <div
                class="name"
              >
                Балльный счет
              </div>
              <div
                class="value"
              >
                aa1HemNi
              </div>
              <div
                class="status neutral"
              >
                Сис имя: nCE33ILt
              </div>
            </div>
          </div>
          <div
            class="half"
          >
            <div
              class="name"
            >
              Сумма
            </div>
            <div
              class="promoValue"
            >
              10465118.7
            </div>
            <div
              class="line"
            >
              <div
                class="name"
              >
                До
              </div>
              <div
                class="value"
              >
                Sun Sep 15 2019
              </div>
            </div>
          </div>
        </div>
        <div
          class="line"
        >
          <span
            class="promoDetail"
          >
            181652215
          </span>
          <span
            class="promoDetail"
          >
            promotion-1
          </span>
          <span
            class="promoDetail"
          >
            mindbox
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="promo"
      >
        <div
          class="promoType"
        >
          Скидка балламми
        </div>
        <div
          class="inline"
        >
          <div
            class="half"
          >
            <div
              class="value"
            >
               Новая акция
            </div>
            <div
              class="line"
            >
              <div
                class="name"
              >
                Примененный промокод
              </div>
              <div
                class="value"
              >
                ABC12345
              </div>
              <div
                class="status neutral"
              >
                Из пула: Новый пул промокодов
              </div>
            </div>
            <div
              class="line"
            >
              <div
                class="name"
              >
                Балльный счет
              </div>
              <div
                class="value"
              >
                dhVKsVFD
              </div>
              <div
                class="status neutral"
              >
                Сис имя: Dc5EuQOG
              </div>
            </div>
          </div>
          <div
            class="half"
          >
            <div
              class="name"
            >
              Сумма
            </div>
            <div
              class="promoValue"
            >
              15785603.89
            </div>
          </div>
        </div>
        <div
          class="line"
        >
          <span
            class="promoDetail"
          >
            355187063
          </span>
          <span
            class="promoDetail"
          >
            promotion-1
          </span>
          <span
            class="promoDetail"
          >
            mindbox
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="promo"
      >
        <div
          class="promoType"
        >
          Скидка
        </div>
        <div
          class="inline"
        >
          <div
            class="half"
          >
            <div
              class="value"
            >
               Новая акция
            </div>
            <div
              class="line"
            >
              <div
                class="name"
              >
                Примененный промокод
              </div>
              <div
                class="value"
              >
                ABC12345
              </div>
              <div
                class="status neutral"
              >
                Из пула: Новый пул промокодов
              </div>
            </div>
          </div>
          <div
            class="half"
          >
            <div
              class="name"
            >
              Сумма
            </div>
            <div
              class="promoValue"
            >
              15872309.02
            </div>
          </div>
        </div>
        <div
          class="line"
        >
          <span
            class="promoDetail"
          >
            1761263450
          </span>
          <span
            class="promoDetail"
          >
            promotion-1
          </span>
          <span
            class="promoDetail"
          >
            mindbox
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="promo"
      >
        <div
          class="promoType"
        >
          Скидка на доставку
        </div>
        <div
          class="inline"
        >
          <div
            class="half"
          >
            <div
              class="value"
            >
               Новая акция
            </div>
            <div
              class="line"
            >
              <div
                class="name"
              >
                Примененный промокод
              </div>
              <div
                class="value"
              >
                ABC12345
              </div>
              <div
                class="status neutral"
              >
                Из пула: Новый пул промокодов
              </div>
            </div>
          </div>
          <div
            class="half"
          >
            <div
              class="name"
            >
              Сумма
            </div>
            <div
              class="promoValue"
            >
              4987163.88
            </div>
          </div>
        </div>
        <div
          class="line"
        >
          <span
            class="promoDetail"
          >
            361022015
          </span>
          <span
            class="promoDetail"
          >
            promotion-1
          </span>
          <span
            class="promoDetail"
          >
            mindbox
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="promo"
      >
        <div
          class="promoType"
        >
          Сообщение
        </div>
        <div
          class="inline"
        >
          <div
            class="half"
          >
            <div
              class="value"
            >
               Новая акция
            </div>
            <div
              class="line"
            >
              <div
                class="name"
              >
                Примененный промокод
              </div>
              <div
                class="value"
              >
                ABC12345
              </div>
              <div
                class="status neutral"
              >
                Из пула: Новый пул промокодов
              </div>
            </div>
          </div>
        </div>
        <div
          class="line"
        >
          <span
            class="promoDetail"
          >
            1991884686
          </span>
          <span
            class="promoDetail"
          >
            promotion-1
          </span>
          <span
            class="promoDetail"
          >
            mindbox
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="promo"
      >
        <div
          class="promoType"
        >
          Выданный промокод
        </div>
        <div
          class="inline"
        >
          <div
            class="half"
          >
            <div
              class="value"
            >
               Новая акция
            </div>
            <div
              class="line"
            >
              <div
                class="name"
              >
                Примененный промокод
              </div>
              <div
                class="value"
              >
                ABC12345
              </div>
              <div
                class="status neutral"
              >
                Из пула: Новый пул промокодов
              </div>
            </div>
            <div
              class="line"
            >
              <div
                class="name"
              >
                Выданный промокод
              </div>
              <div
                class="value"
              >
                ABC12345
              </div>
              <div
                class="status neutral"
              >
                Из пула:Новый пул промокодов
              </div>
            </div>
          </div>
        </div>
        <div
          class="line"
        >
          <span
            class="promoDetail"
          >
            1303715822
          </span>
          <span
            class="promoDetail"
          >
            promotion-1
          </span>
          <span
            class="promoDetail"
          >
            mindbox
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="promo"
      >
        <div
          class="promoType"
        >
          Скидка
        </div>
        <div
          class="inline"
        >
          <div
            class="half"
          >
            <div
              class="value"
            >
               Новая акция
            </div>
          </div>
          <div
            class="half"
          >
            <div
              class="name"
            >
              Сумма
            </div>
            <div
              class="promoValue"
            >
              2714279.45
            </div>
          </div>
        </div>
        <div
          class="line"
        >
          <span
            class="promoDetail"
          >
            1507016125
          </span>
          <span
            class="promoDetail"
          >
            promotion-1
          </span>
          <span
            class="promoDetail"
          >
            mindbox
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="promo"
      >
        <div
          class="promoType"
        >
          Скидка
        </div>
        <div
          class="inline"
        >
          <div
            class="half"
          >
            <div
              class="line"
            >
              <div
                class="name"
              >
                Примененный промокод
              </div>
              <div
                class="value"
              >
                ABC12345
              </div>
              <div
                class="status neutral"
              >
                Из пула: Новый пул промокодов
              </div>
            </div>
          </div>
          <div
            class="half"
          >
            <div
              class="name"
            >
              Сумма
            </div>
            <div
              class="promoValue"
            >
              20439998.48
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
